src/main/java/com/epam/prejap/teatrees/game/Grid.java

```diff
@@ -0,0 +1,94 @@
+package com.epam.prejap.teatrees.game;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+/**
+ * Provides Playfield an api for working with 2D array.
+ * @author dominik_kaminski
+ * @author przemyslaw_szewczyk
+ */
+record Grid(byte[][] grid, int rows, int cols) {
+    Grid(int rows, int cols) {
+        this(new byte[rows][cols], rows, cols);
+    }
+
+    Grid(byte[][] grid) {
+        this(grid, grid.length, grid[0].length);
+    }
+
+    int cellAt(int row, int col) {
+        return grid[row][col];
+    }
+
+    void setCell(int row, int col, int value) {
+        grid[row][col] = (byte) value;
+    }
+
+    void removeCompleteLines() {
+        for (int i = rows - 1; i >= 0; i--)
+        {
+            if (isRowComplete(i))
+            {
+                dropRows(i - 1);
```

Comment:
dropRows() should point at exactly same number like in the method before it.
i-1 is confusing.
to change it in dropRows you can change while loop like it will be till i>=1
(But you know I'm not sure 'hahaha')

Reply:
Line removing algo has been changed and is more understandable (I hope ;p).